contracts/v2/marketplace/TokenMarketplaceV2.sol

```diff
@@ -0,0 +1,496 @@
+
+  // Explicitly disable sales for specific tokens
+  mapping(uint256 => bool) public disabledTokens;
+
+  ///////////////
+  // Modifiers //
+  ///////////////
+
+  modifier onlyWhenOfferOwner(uint256 _tokenId) {
+    require(offers[_tokenId].bidder == msg.sender, "Not offer maker");
+    _;
+  }
+
+  modifier onlyWhenTokenExists(uint256 _tokenId) {
+    require(kodaAddress.exists(_tokenId), "Token does not exist");
+    _;
+  }
+
+  modifier onlyWhenBidOverMinAmount(uint256 _tokenId) {
+    require(msg.value >= offers[_tokenId].offer.add(minBidAmount), "Offer not enough");
+    _;
+  }
+
+  modifier onlyWhenTokenAuctionEnabled(uint256 _tokenId) {
+    require(!disabledTokens[_tokenId], "Token not enabled for offers");
+    _;
+  }
+
+  /////////////////
+  // Constructor //
+  /////////////////
+
+  // Set the caller as the default KO account
+  constructor(IKODAV2Methods _kodaAddress, address _koCommissionAccount) public {
+    kodaAddress = _kodaAddress;
+    koCommissionAccount = _koCommissionAccount;
+    super.addAddressToWhitelist(msg.sender);
+  }
+
+  //////////////////////////
+  // User Bidding Actions //
+  //////////////////////////
+
+  function placeBid(uint256 _tokenId)
+  public
+  payable
+  whenNotPaused
+  nonReentrant
+  onlyWhenTokenExists(_tokenId)
+  onlyWhenBidOverMinAmount(_tokenId)
+  onlyWhenTokenAuctionEnabled(_tokenId)
+  {
+    require(!isContract(msg.sender), "Unable to place a bid as a contract");
+    _refundHighestBidder(_tokenId);
+
+    offers[_tokenId] = Offer(msg.sender, msg.value);
+
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+
+    emit BidPlaced(_tokenId, currentOwner, msg.sender, msg.value);
+  }
+
+  function withdrawBid(uint256 _tokenId)
+  public
+  whenNotPaused
+  nonReentrant
+  onlyWhenTokenExists(_tokenId)
+  onlyWhenOfferOwner(_tokenId)
+  {
+    _refundHighestBidder(_tokenId);
+
+    emit BidWithdrawn(_tokenId, msg.sender);
+  }
+
+  function rejectBid(uint256 _tokenId)
+  public
+  whenNotPaused
+  nonReentrant
+  {
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+    require(currentOwner == msg.sender, "Not token owner");
+
+    uint256 currentHighestBiddersAmount = offers[_tokenId].offer;
+    require(currentHighestBiddersAmount > 0, "No offer open");
+
+    address currentHighestBidder = offers[_tokenId].bidder;
+
+    _refundHighestBidder(_tokenId);
+
+    emit BidRejected(_tokenId, currentOwner, currentHighestBidder, currentHighestBiddersAmount);
+  }
+
+  function acceptBid(uint256 _tokenId, uint256 _acceptedAmount)
+  public
+  whenNotPaused
+  nonReentrant
+  {
+    address currentOwner = kodaAddress.ownerOf(_tokenId);
+    require(currentOwner == msg.sender, "Not token owner");
+
+    Offer storage offer = offers[_tokenId];
+
+    uint256 winningOffer = offer.offer;
+
+    // Check valid offer and offer not replaced whilst inflight
+    require(winningOffer > 0 && _acceptedAmount >= winningOffer, "Offer amount not satisfied");
```

Comment:
why would you pass more - I guess incase a higher bid comes in while inflight?

Reply:
yep